pkg/collector/corechecks/servicediscovery/servicediscovery.go

```diff
@@ -119,19 +120,26 @@ func newCheck() check.Check {
 
 // Configure parses the check configuration and initializes the check
 func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, instanceConfig, initConfig integration.Data, source string) error {
-	if !pkgconfig.Datadog().GetBool("service_discovery.enabled") {
-		return errors.New("service discovery is disabled")
+	if !ddconfig.Datadog().GetBool("service_discovery.enabled") {
+		return errors.New("service_discovery check is disabled")
 	}
 	if newOSImpl == nil {
 		return errors.New("service_discovery check not implemented on " + runtime.GOOS)
 	}
+	if ddconfig.IsContainerized() {
+		return errors.New("service_discovery check is not supported in containerized environments")
```

Comment:
This isn't an error condition, I think. I would expect we would just exit and maybe log at debug level.

Reply:
Seeing how other checks are done, this `Configure` function is supposed to be used like this (e.g. return an error when the check is disabled from config).

In theory, users won't see this error because we make sure the check is not added to the list to initialize under this condition, but if for some reason they get here, they will just see 1 warn log.